### ISSUE TITLE
cli-plugin-pwa: stop pre-caching files generated by route level code splitting

### ISSUE THREAD

**Issue by bobbybol** [vuejs/vue-cli#3332]
### What problem does this feature solve?
We can split code at route level, so the client doesn't have to download one monolithic app, but instead load smaller parts upon request. This good practice is nullified however when using the pwa plugin, because all generated js files go into the precache manifest, hence we're back to downloading the whole application on first load. A proposed feature would be to leave out the 'split' js files, so the end user can have the PWA experience without long initial load times.

### What does the proposed API look like?
No API changes.

**Comment by LinusBorg** [comment]
> This good practice is nullified however when using the pwa plugin, because all generated js files go into the precache manifest, hence we're back to downloading the whole application on first load. 

That's incorrect. These files will be downloaded in the background after the initial load, when network capacity etc. allow it. So it won't have an influence on TTI etc.

See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Link_prefetching_FAQ

It's a recommended practice for PWAs: https://medium.com/reloading/preload-prefetch-and-priorities-in-chrome-776165961bbf

However if you don't want it: 
```javascript
chainWebpack: config => {
  config.plugins.delete('prefetch')
}
```

**Comment by bobbybol** [comment]
Fantastic, thank you Thorsten for these articles, I learned something today ;)
I will do more experiments, but I would still maintain that the best practice is to mark "resources likely to be visited" as precache. The current PWA plugin adds _everything_ to resources to be prefetched (including large images and videos). More control over what to prefetch and what to cache on visit (basically excluding those resource from precache manifest and doing custom cache logic in your sw) could still be a feasible feature request.

**Comment by LinusBorg** [comment]
Oh i just realizeedd that i partly missunderstood, i thought you were indeed talking about pre*fetched* chunks only but you mean pre*cached* stuff which is more. 

You should check the readme for the work box plugin which the pwa option is using. It has an `exclude` option
